experimental/GModule/Cohomology.jl

```diff
@@ -184,11 +184,12 @@ end
 function fp_group_with_isomorphism(C::GModule)
   #TODO: better for PcGroup!!!
   if !isdefined(C, :F)
-    if (!isa(group(C), FPGroup)) && is_trivial(group(C))
+    G = group(C)
+    if (!isa(group(C), FPGroup)) && is_trivial(G)
```

Comment:
Why are you not using `G` here and in the next lines? Is it true that
 - G always has no gens OR
 - any hom G->X can be defeined using no gens?

Reply:
If G is trivial then there is exactly one homomorphism from G to another group, and `hom` creates it when we give it two empty lists, independent of how many generators G actually has.
In fact using `gens(C)` caused a problem, because it might be nonempty.

(And yes, `G` should be used instead of `group(C)`; I do not understand how this got lost, the idea was that `group(C)` was now called too often.)